### DISCUSSION TITLE
Possibility of declaritively defining applications?

### DISCUSSION THREAD

**Discussion by btrepp** [goauthentik/authentik#2643] [Ideas]
Hi All,

Absolutely been loving authentik, it's super easy to get going and have my own identities.
At the moment it seems very UI driven to use, this is great for getting started, but does put a lot of care in the postgres database. If that goes all my configuration does too :).

I've been driving most of my homelab (running via kubernetes) to be more declarative, e.g configmaps in git, secrets in passwordstore.org that ansible will inject into the cluster. This means the entire thing can be rebuilt rather painlessly from disaster (or me breaking it). I think the short term solution is to get my PGSQL backup progress down, but it's an extra thing to check.

It would be great if Application/Provider definitions could be provided from a file, then they could be volume mounted as k8s secrets. This would let me configure them however I want, and the configuration would be git friendly.

I know this would be a big feature request, so no expectations that this would happen, just opening discussion in case anyone else would like this (or maybe even knows how to do it now). I was thinking I might be able to achieve this with a pod that is configuring authentik via an API/cli

**Comment by BeryJu** [comment]
Hi, currently the best way to accomplish this is with the authentik terraform provider (https://registry.terraform.io/providers/goauthentik/authentik/latest/docs), but I'm still planning to make this possible in authentik itself (https://github.com/goauthentik/authentik/pull/1573), allowing users to share flows and providers and more, defined as yaml

**Comment by fabianoengler** [comment]
Hey, I landed here via google.

In case more people are searching for this, it looks like this is already fully implemented via the new Blueprints system:
https://docs.goauthentik.io/docs/customize/blueprints/

Looks awesome, great work!
